core/templates/domain/admin/admin-backend-api.service.spec.ts

```diff
@@ -198,3 +199,116 @@ describe('Admin backend api service', () => {
     })
   );
 });
+
+describe('Admin Backend API service for Jobs Tab', () => {
+  let adminBackendApiService: AdminBackendApiService;
+  let httpTestingController: HttpTestingController;
+  let csrfService: CsrfTokenService = null;
+  let successHandler = null;
+  let failHandler = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [AdminBackendApiService]
+    });
+    httpTestingController = TestBed.get(HttpTestingController);
+    adminBackendApiService = TestBed.get(AdminBackendApiService);
+    csrfService = TestBed.get(CsrfTokenService);
+    successHandler = jasmine.createSpy('success');
+    failHandler = jasmine.createSpy('fail');
+
+    spyOn(csrfService, 'getTokenAsync').and.callFake(() => {
+      return Promise.resolve('sample-csrf-token');
+    });
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+  it('should start the new job with respect to its jobType',
+    fakeAsync(() => {
+      let jobType = 'ActivityContributorsSummaryOneOffJob';
+      adminBackendApiService.startNewJob(jobType)
+        .then(successHandler, failHandler);
+
+      let req = httpTestingController.expectOne(
+        '/adminhandler');
+      expect(req.request.method).toEqual('POST');
+      req.flush(200);
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalled();
+      expect(failHandler).not.toHaveBeenCalled();
+    }
+    ));
+
+  it('should cancel the job with respects to its jobType and jobId',
```

Comment:
```suggestion
  it('should cancel the job given its id and type',
```

Reply:
Also, what happens if you try to cancel a job that hasn't started yet? Can we add that as a test case?

Reply:
We can not cancel a job that hasn't started. Our UI only shows up the cancel button when we start a job.